### PR TITLE
feat: show fee when upgrading implementation

### PR DESCRIPTION
**Pull request by simonheys**
Changes the account upgrade action so that it invokes transaction review. Removes previous upgrade method.

Adds local tools and instructions for setting up devnet, declaring contract and upgrading locally.

Recording shows changing the contract class hash on localhost devnet, then upgrading the account via transaction review with fees.

https://user-images.githubusercontent.com/175607/177184964-f468ea92-a035-479f-a1fb-f362c1c3f6c1.mov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +32,10 @@
     "test": "lerna run test --stream",
     "test:e2e": "lerna run test:e2e --stream",
     "prepare": "yarn --cwd packages/get-starknet build && husky install",
-    "storybook": "cd packages/storybook && yarn storybook"
+    "storybook": "cd packages/storybook && yarn storybook",
+    "devnet:start": "docker run -it -p 5050:5050 shardlabs/starknet-devnet:latest --seed 0 --accounts 1",
+    "devnet:start:arm": "docker run -it -p 5050:5050 shardlabs/starknet-devnet:latest-arm --seed 0 --accounts 1",
```
Comment (simonheys) [inline package.json:37]: Would be nice to auto-detect this, but couldn't find a neat way

File: scripts/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      {
+        vars: "all",
+        ignoreRestSiblings: true,
+        argsIgnorePattern: "^_",
+      },
+    ],
+    "@typescript-eslint/no-non-null-assertion": "error",
+    curly: "error",
+  },
+}
```
Comment (simonheys) [inline scripts/.eslintrc.js:31]: pre-commit lint fails without this file

File: scripts/devnet/devnet-upgrade-helper.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  stark,
+  uint256,
+  validateAndParseAddress,
+} from "starknet"
+
+/** Fee token address on devnet */
+const feeTokenContractAddressOnlyForLocalTesting =
+  "0x6a7a6243f92a347c03c935ce4834c47cbd2a951536c10319168866db9d57983"
+
+/** NOTE: these values are from passing --seed 0 --accounts 1 to starkent-devenet and are ONLY for local testing */
```
Comment (simonheys) [inline scripts/devnet/devnet-upgrade-helper.ts:19]: Wanted to find a way to parse these from Docker output, but too contrived

File: scripts/.eslintrc.js (hunk 2 above)
Comment (janek26) [reply scripts/.eslintrc.js:31]: we could also ignore that folder

File: package.json [hunk 4]
```diff
@@ -5,13 +5,18 @@
...
+    "@tsconfig/node18": "^1.0.1",
+    "@types/prompt": "^1.1.2",
     "bundlewatch": "^0.3.3",
     "husky": "^8.0.0",
     "import-sort-style-module": "^6.0.0",
     "lerna": "^5.0.0",
     "lint-staged": ">=10",
     "prettier": ">=2",
     "prettier-plugin-import-sort": "^0.0.7",
+    "prompt": "^1.3.0",
```
[the remark this answers is not in the document]
Comment (janek26) [reply package.json:17]: why is it needed? where is it used

File: packages/extension/e2e/apis/sendDevnetEthToAccount.ts [hunk 5]
```diff
@@ -1,16 +1,6 @@
...
-const pk = "0xe3e70682c2094cac629f6fbed82c07cd" // put hex representation of private key here
-
-const provider = new Provider({ baseUrl: "http://127.0.0.1:5050/" }) // connect to devnet
-const keyPair = ec.getKeyPair(pk)
-const account = new Account(
-  provider,
-  "0x7e00d496e324876bbc8531f2d9a82bf154d1a04a50218ee74cdd372f75a551a".toLowerCase(), // put account address which is controlled by the private key here
-  keyPair,
-)
+import { account } from "./account"
```
Comment (simonheys) [inline packages/extension/e2e/apis/sendDevnetEthToAccount.ts:3]: Small refactor to use shared account

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1 @@
+scripts
```
Comment (simonheys) [inline .eslintignore:1]: Prevents linting error on precommit as discussed previously

**Review by janek26 — APPROVED** [review]
looks good! e2e test would be cool at some point

**Pull request merged**